### PR TITLE
[2/X] [1/Y] Move insertion into index to event creation.

### PR DESCRIPTION
**Pull request by nikhitashanker**
- Move insertion into index to event creation such that SearchDataServlet is only using to retrieve results.
- Check order of results produced by SearchDataServlet as expected with the optimized algorithm.

### CODE REVIEW

File: project/src/test/java/com/google/sps/EventSpannerTasksTest.java [hunk 1]
```diff
@@ -139,7 +148,8 @@ public void testEventCreationDoPost() throws Exception {
         .setEnvEmail(EMAIL)
         .setEnvAuthDomain("example.com");
 
-    new EventCreationServlet().doPost(request, response);
+
```
Comment (dash102) [inline project/src/test/java/com/google/sps/EventSpannerTasksTest.java:151]: nit: take out extra newline

File: project/src/test/java/com/google/sps/SearchDataServletTest.java [hunk 2]
```diff
@@ -25,27 +42,86 @@
   private HttpServletResponse secondPostResponse;
   private HttpServletRequest getRequest;
   private HttpServletResponse getResponse;
-  private StringWriter stringWriter;
-  private PrintWriter printWriter;
+  private StringWriter getStringWriter;
```
Comment (dash102) [inline project/src/test/java/com/google/sps/SearchDataServletTest.java:45]: I don't entirely understand why there are multiple StringWriters and PrintWriters - I don't really see where they're used

Comment (nikhitashanker) [reply project/src/test/java/com/google/sps/SearchDataServletTest.java:45]: Ah so in the EventCreationServlet, the Servlet calls getWriter() so if I dont use these writers getWriter() will return null.

File: project/src/test/java/com/google/sps/EventSpannerTasksTest.java (hunk 1 above)
Comment (nikhitashanker) [reply project/src/test/java/com/google/sps/EventSpannerTasksTest.java:151]: Done.

File: project/src/test/java/com/google/sps/SearchDataServletTest.java [hunk 3]
```diff
@@ -25,27 +42,83 @@
...
-  private static final String DESCRIPTION_WITHOUT_CAKE_KEYWORD =
-    "Sutter Middle School will be walking to McKinley Park. 7th grade class and teachers"
-    + " will have a picnic, play games and eat lunch at the park and Clunie Pool.";
+  private static final String GAMES = "games";
+  private static final String NAME_WITHOUT_GAMES = "End of the Year Picnic";
+  private static final String DESCRIPTION_WITHOUT_GAMES =
+      "Sutter Middle School will be walking to McKinley Park. 7th grade class and teachers will"
+          + " have a picnic and eat lunch at the park and Clunie Pool.";
+  private static final String NAME_WITH_GAMES = "End of the Year Picnic and Games";
+  private static final ArrayList<Keyword> KEYWORDS_NAME_WITHOUT_GAMES =
```
Comment (yicyao) [inline project/src/test/java/com/google/sps/SearchDataServletTest.java:67]: HIGH_RELEVANCE ? 
I guess my question with these tests is like KEYWORDS_NAME_WITHOUT_GAMES (with no relevance labeled) 'medium relevance' or does the relevance not matter in these?

File: project/src/test/java/com/google/sps/SearchDataServletTest.java [hunk 4]
```diff
@@ -58,45 +131,98 @@ public void setUp() throws Exception {
...
+
+  @After
+  public void tearDown() throws Exception {
+    SpannerTestTasks.cleanup();
+    authenticationHelper.tearDown();
   }
 
   @Test
-  public void addEventAndRetrieveResultsForKeywordNotInEventDescription_noResultsReturned()
+  public void oneEvent_KeywordNotRelevantInEventTitleOrDescription_noResultsReturned()
```
Comment (yicyao) [inline project/src/test/java/com/google/sps/SearchDataServletTest.java:168]: @hlz03 @liangqigong a best practices question: generally would it be better for tests to name our tests descriptively like this vs writing a comment to describe it?

Comment (liangqigong) [reply project/src/test/java/com/google/sps/SearchDataServletTest.java:168]: I think this name should be fine. And if you think it is to hard describe it in a few words, a comment is also acceptable.

**Comment by nikhitashanker** [comment]
Resolved merge conflicts.

File: project/src/test/java/com/google/sps/SearchDataServletTest.java [hunk 5]
```diff
@@ -58,45 +143,120 @@ public void setUp() throws Exception {
...
+    setRequiredRequestParameters(postRequest);
     Mockito.when(getRequest.getParameter(PARAMETER_KEYWORD)).thenReturn(CAKE);
 
-    searchDataServlet.doPost(postRequest, postResponse);
+    // On the first call the getKeywords for title, return empty list
+    // On the second call the getKeyords for description, return empty list
+    Mockito.when(mockKeywordHelper.getKeywords())
+        .thenReturn(KEYWORDS_NAME_WITHOUT_GAMES, KEYWORDS_DESCRIPTION_WITHOUT_GAMES);
+
+    eventCreationServlet.doPost(postRequest, postResponse);
```
Comment (liangqigong) [inline project/src/test/java/com/google/sps/SearchDataServletTest.java:206]: Add comments for this part the the following part.

Comment (nikhitashanker) [reply project/src/test/java/com/google/sps/SearchDataServletTest.java:206]: Done.

**Pull request merged**